articles/tokens/refresh-token.md

```diff
@@ -1,152 +1,173 @@
+A refresh token allows the application to request Auth0 to issue a new `access_token` or `id_token` directly, without needing to re-authenticate the user. This will work as long as the refresh token has not been revoked.
 
-Refresh tokens can be issued for each combination of __app__, __user__ and __device__. Once the Auth0 refresh token is issued, the values of the client, user, and device set during its creation cannot be changed.
-
-::: panel-warning Secure Storage
+::: panel-warning Security Warning
 Refresh tokens must be stored securely by an application since they allow a user to remain authenticated essentially forever.
 :::
 
-Refresh tokens can be [obtained](#obtain-a-refresh-token) or [revoked](#revoke-a-refresh-token-using-the-management-api) programmatically through the Auth0 API.
+Refresh tokens can be [obtained](#obtain-a-refresh-token) or [revoked](#revoke-a-refresh-token-with-a-request) programmatically through the Auth0 API.
 
-They can also be viewed and revoked [from the dashboard](#revoke-a-refresh-token-in-the-dashboard).
+They can also be viewed and revoked [from the dashboard](#revoke-a-refresh-token-using-the-dashboard).
 
-## Obtain a Refresh Token
+## Get a Refresh Token
 
-To obtain a refresh token, the `offline_access` scope (see: [Scopes](/scopes)) and an arbitrary `device` name must be included when initiating an authentication request through the [authorize](/api/authentication/reference#authorize-client) endpoint.
+To get a refresh token, you must include the `offline_access` [scope](/scopes) when you initiate an authentication request through the [authorize](/api/authentication/reference#authorize-client) endpoint.
 
 For example:
 
-```text
+```
 GET https://${account.namespace}/authorize/?
     response_type=token
+    &audience=YOUR_API_IDENTIFIER
     &client_id=${account.clientId}
-    &redirect_uri=YOUR_CALLBACK_URL
+    &redirect_uri=${account.callback}
     &state=VALUE_THAT_SURVIVES_REDIRECTS
     &scope=openid%20offline_access
-    &device=my-device
 ```
 
-**NOTE**: The `device` parameter can be any value, such as a unique mobile device identifier.
-
-When the authentication flow completes, Auth0 will redirect the user to the `callback_URL` as usual.
+Once the user authenticates successfully, the client will be redirected to the `callback_URL`.
 The complete URL will be as follows:
 
-```text
-GET https://YOUR_CALLBACK_URL#
+```
+GET ${account.callback}#
     access_token=2nF...WpA
     &id_token=eyJhb...
     &state=VALUE_THAT_SURVIVES_REDIRECTS
     &refresh_token=Cqp...Mwe
 ```
 
-The refresh token is returned as part of the URL, in the form of an opaque string.
+The refresh token (an opaque string) is part of the URL. You should store it securely and use it only when needed.
+
+**NOTE**: In this example, the token was returned to the client in the URL because the [implicit grant](/api-auth/grant/implicit) (`response_type=token`) was used.
```

Comment:
The implicit grant cannot issue refresh tokens. Only Authorization Code, PKCE and password grant.